certbot/interfaces.py

```diff
@@ -238,6 +238,12 @@ class IConfig(zope.interface.Interface):
         "This only affects the port Certbot listens on. "
         "A conforming ACME server will still attempt to connect on port 80.")
 
+    http01_use_tls = zope.interface.Attribute(
+        "Runs the standalone server with a self signed certificate. "
+        "Useful for when you need to validate an http-01 challenge with "
+        "an https endpoint or if you have an upstream proxy redirecting "
+        "all http traffic to https.")
```

Comment:
Oh, in that case, I might have misunderstood the purpose of this PR (or at least some of it). I assumed this was for something like a CDN that enforced TLS for backend connections, where the CDN itself still listens and responds on port 80.

`http-01` always works via HTTP on port 80, at least for the initial request. HTTP redirects are allowed. HTTPS-only hosts would fail validation. Here's the relevant spec section:

>    As a domain may resolve to multiple IPv4 and IPv6 addresses, the
   server will connect to at least one of the hosts found in A and AAAA
   records, at its discretion.  **Because many webservers allocate a
   default HTTPS virtual host to a particular low-privilege tenant user
   in a subtle and non-intuitive manner, the challenge must be completed
   over HTTP, not HTTPS.**
> https://tools.ietf.org/html/draft-ietf-acme-acme-01#section-7.2